src/click/utils.py

```diff
@@ -474,8 +474,11 @@ def __getattr__(self, attr: str) -> t.Any:
         return getattr(self.wrapped, attr)
 
 
+_unset = ModuleType("_click_unset")
+
+
 def _detect_program_name(
-    path: t.Optional[str] = None, _main: ModuleType = sys.modules["__main__"]
+    path: t.Optional[str] = None, _main: ModuleType = _unset
```

Comment:
Looks like `_main=None` is being passed in by a test. But if I fix the type annotation, then I get an error for accessing an attribute of None. Without understanding the logic of the function, I get the feeling that either the test is wrong of the type annotation is wrong. If it's the former, then fixing it in this MR makes sense because the default can then be None (instead of a faux module). If it's the latter, I can do it in this PR if you need me to or punt it for a future PR.